nautobot/core/context_processors.py

```diff
@@ -47,6 +49,34 @@ def nav_menu(request):
     Expose nav menu data for navigation and global search.
     Also, indicate whether `"nautobot_version_control"` app is installed in order to render branch picker in nav menu.
     """
+    has_identified_active_link = False
+    related_list_view_link = None
+    if request.resolver_match:
+        # Try to map requested page `view_name` to a specific `model` via `lookup.get_model_for_view_name`.
+        try:
+            model = lookup.get_model_for_view_name(request.resolver_match.view_name)
+        except ValueError:
+            model = None
+
+        # If model mapping above fails, fall back to deriving a `model` from requested page `view_class` `queryset`.
+        if not model:
+            view_func = request.resolver_match.func
+            view_class = None
+            if hasattr(view_func, "view_class"):
+                view_class = view_func.view_class
+            elif hasattr(view_func, "cls"):
+                view_class = view_func.cls
```

Comment:
IIRC this logic is handling the difference between a View and a ViewSet? If that's accurate, an explanatory comment might be nice.